### PR TITLE
Add Antiaffinity & PDBs to dns-resolver, Kubermatic and the monitoring components

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

Adds Pod Antiaffinity and PDBs to the dns-resolver, the Kubermatic control plane and the monitoring components.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Pod AntiAffinity and PDBs were added to the Kubermatic control plane components and the monitoring stack to spread them out if possible and reduce the chance of unavailability
```

### CODE REVIEW

File: config/kubermatic/templates/kubermatic-api-dep.yaml [hunk 1]
```diff
@@ -5,6 +5,15 @@ metadata:
   name: kubermatic-api-v1
 spec:
   replicas: {{ .Values.kubermatic.api.replicas }}
+  affinity:
```
Comment (xrstf) [inline config/kubermatic/templates/kubermatic-api-dep.yaml]: Deployment spec's have affinity fields? https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#deploymentspec-v1-apps does not mention them.

Also, affinities are meant to be configured in the chart's values.yaml, same for all other charts and we already have *node*AntiAffinities for `kubernetes.io/hostname` -- are those equivalent to your podAntiAffinity?

**Review by xrstf — CHANGES_REQUESTED** [review]

**Comment by alvaroaleman** [comment]
> Deployment spec's have affinity fields? https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#deploymentspec-v1-apps does not mention them.

Ofc not :man_facepalming: Good catch!

> Also, affinities are meant to be configured in the chart's values.yaml, same for all other charts 

Yeah because I was looking for affinities at the wrong place I didn't see that. Fixed it up to be in the `values.yaml` everywhere.

> and we already have nodeAntiAffinities for kubernetes.io/hostname -- are those equivalent to your podAntiAffinity?

I think what you mean is `nodeAffinity` for `kubermatic.io/type`? Thats the only affinity I could find.

But those two are not the same. `nodeAffinity` is used to (not-) co-locate a set of pod on a set of nodes whereas `podAntiAffinity` is used to (not-) co-locate a set of pods with each other. Most of the times the latter is done with the purpose of spreading them out in order to reduce the chance of unavailability when a node fails.

Both of them have a `weight`, I have reduced the `weight` of the `podAntiAffinities` to `10` in all places where there was `nodeAffinity` to make sure the `nodeAffinity` is respected first. Let me know what you think.

File: config/kubermatic/templates/kubermatic-controller-manager-dep.yaml [hunk 2]
```diff
@@ -4,6 +4,7 @@ metadata:
   name: controller-manager-v1
 spec:
   replicas: {{ .Values.kubermatic.controller.replicas }}
+  affinity:
```
Comment (xrstf) [inline config/kubermatic/templates/kubermatic-controller-manager-dep.yaml]: Leftover?

**Review by xrstf — CHANGES_REQUESTED** [review]

**Comment by alvaroaleman** [comment]
@xrstf Fixed the leftover

**Review by xrstf — APPROVED** [review]

**Pull request merged**